Ukol_3/cpp/linked_list.cpp

```diff
@@ -0,0 +1,212 @@
+    Node* temp = *head;
+    for (int i = 0; i < index - 1 && temp != nullptr; i++) {
+        temp = temp->next;
+    }
+    if (temp == nullptr) {
+        std::cout << "Index je mimo rozsah\n";
+        delete newNode;
+        return;
+    }
+    newNode->next = temp->next;
+    temp->next = newNode;
+}
+
+// Funkce pro smazání uzlu ze začátku seznamu
+void deleteAtBeginning(Node** head)
+{
+    Node* temp = (*head)->next;
+    delete* head;
+    *head = temp;
+
+}
+
+// Funkce pro smazání uzlu z konce seznamu
+void deleteAtEnd(Node** head)
+{
+    if ((*head)->next == nullptr) {
+        deleteAtBeginning(head);
+        return;
+
+    }
+    Node* posledni;
+    Node* predposledni;
+    posledni = *head;
+    while (posledni->next != nullptr) {
+        posledni = posledni->next;
+
+    }
+    predposledni = *head;
+    while (predposledni->next->next != nullptr) {
+        predposledni = predposledni->next->next;
+
+    }
+
+    delete posledni;
+    predposledni->next = nullptr;
+
+
+}
+
+// Funkce pro smazani uzlu na indexu
+void deleteAtIndex(Node* head, int index)
+{
+    if (head == nullptr) {
+        return;
+    }
+
+    if (index == 0) {
+        deleteAtBeginning(&head);
+        return;
+
+    }
+    Node* aktualni = head;      //ukazatel, ktery zacina na prvnim uzlu (head), ten budeme mazat
+    Node* predchozi = nullptr;      //bude slouzit k uchovani predchoziho uzlu
+    int aktualnipozice = 0;
+
+    while (aktualni != nullptr && aktualnipozice < index) {
+        predchozi = aktualni;       //aktualizujeme, aby predchozi ukazoval na aktualni uzel
+        aktualni = aktualni->next;
+        aktualnipozice++;
+
+    }
+    if (aktualni != nullptr) {
+        predchozi->next = aktualni->next;   //vynechavani uzlu, ktery chceme smazat
+        delete aktualni;
+
+    }
+
+}
+
+// Funkce pro nalezeni prvniho vyskytu
+int findFirstOccurrence(Node* head, int value)
+{
+        int index = 0;
+        while (head != nullptr) {
+            if (head->data == value) {
+                return index;
+            }
+            head = head->next;
+            index++;
+        }
+
+        return -1;
+
+}
+
+// Funkce pro třídění seznamu
+void sortList(Node** head)
+{
+    if (*head == nullptr) return;
+
+    bool swapped;
+    do {
+        swapped = false;
+        Node* current = *head;
+        while (current->next != nullptr) {
+            if (current->data > current->next->data) {
+                std::swap(current->data, current->next->data);
+                swapped = true;
+            }
+            current = current->next;
+        }
+    } while (swapped);
+}
+
+// Funkce pro smazani (dealokaci) seznamu
+void deleteList(Node** head)
+{
+    while (*head != nullptr) {
+        deleteAtBeginning(head);
+    }
+}
+
+// Operátor pro tisk dat
+std::ostream& operator<<(std::ostream& os, Node* head)
+{
+    Node* current = head;
+    while (current != nullptr) {
+        os << current->data << " ";
```

Comment:
Zde je to teměř v pořádku jen aby prošli testy nesmí byt mezera po každém znaku, ale pouze mezi nimi. Tedy "1 2 3" ne "1 2 3 "
Body za třetí úkol vám dám